docs/arch/02-core-concepts.md

```diff
@@ -0,0 +1,693 @@
+
+**Session types:**
+
+1. **SSE Session**: For stdio transport with SSE proxy mode
+   - Tracks connected SSE clients (multiple clients can connect, but share single stdio connection to container)
+   - Message queue per client
+   - Endpoint URL generation
+   - Note: stdio transport has single connection/session to container
+
+2. **Streamable Session**: For stdio transport with streamable proxy mode
+   - Tracks `Mcp-Session-Id` header
+   - Request/response correlation
+   - Ephemeral sessions for sessionless requests
+
+3. **Proxy Session**: For transparent proxy (SSE/Streamable transports)
+   - Session ID detection from headers
+   - Session ID detection from SSE body
+   - Minimal state tracking
+
+**Session lifecycle:**
+- Created on first request or explicit initialize
+- Tracked via session manager with TTL
+- Cleaned up after inactivity or explicit deletion
+
+**Implementation:**
+- Manager: `pkg/transport/session/manager.go`
+- SSE: `pkg/transport/session/sse_session.go`
+- Streamable: `pkg/transport/session/streamable_session.go`
+- Proxy: `pkg/transport/session/proxy_session.go`
+
+**Related concepts:** Transport, Proxy
+
+### Runtime
+
+A **runtime** is an abstraction over container orchestration systems. It provides a unified interface for container operations.
+
+**Runtime types:**
+
+1. **Docker Runtime**: Docker Engine API
+2. **Podman Runtime**: Podman socket API
+3. **Colima Runtime**: Docker-compatible (uses Docker runtime)
+4. **Kubernetes Runtime**: Kubernetes API (StatefulSets)
+
+**Runtime interface:**
+- `DeployWorkload` - Create and start workload
+- `StopWorkload` - Stop workload
+- `RemoveWorkload` - Delete workload
+- `ListWorkloads` - List all workloads
+- `GetWorkloadInfo` - Get workload details
+- `GetWorkloadLogs` - Retrieve logs
+- `AttachToWorkload` - Attach to stdin/stdout (stdio only)
+- `IsWorkloadRunning` - Check if running
+
+**Runtime detection:**
+Order: Podman → Colima → Docker → Kubernetes (via env)
+
+**Implementation:**
+- Interface: `pkg/container/runtime/types.go`
+- Factory: `pkg/container/factory.go`
+- Detection: `pkg/container/runtime/types.go`
+
+**Related concepts:** Deployer, Workload, Container
+
+### Client
+
+An **MCP client** is an application that uses MCP servers (e.g., Claude Desktop, IDEs, AI tools).
+
+**Client types:**
+- `claude-code` - Claude Code
+- `cursor` - Cursor editor
+- `vscode` - VS Code
+- `cline` - Cline extension
+- `windsurf` - Windsurf editor
+- Many more...
+
+**Client configuration:**
+ToolHive can automatically configure clients to use MCP servers:
+- Reads client config files
+- Adds server URLs
+- Updates on workload start/stop
+- Supports multiple config formats
+
+**Client operations:**
+- `thv client list` - Show available clients
+- `thv client setup <client>` - Configure client
+- `thv client status` - Show client status
+
+**Implementation:**
+- Client types: `pkg/client/types.go`
+- Manager: `pkg/client/manager.go`
```

Comment:
File path reference is incorrect: `pkg/client/types.go` does not exist.

Client types are defined in:
- `pkg/client/config.go`
- `pkg/client/manager.go`

Consider updating to reference `pkg/client/config.go` for client types.

Evidence: Verified via codebase exploration